pinot-core/src/main/java/com/linkedin/pinot/core/query/scheduler/QueryScheduler.java

```diff
@@ -134,17 +134,28 @@ public void stop() {
     byte[] responseData = serializeDataTable(queryRequest, dataTable);
 
     // Log the statistics
+    long numDocsScanned = 0;
+    long numDocsScannedInFilter = 0;
+    long numDocsScannedPostFilter = 0;
+    try {
+      numDocsScanned = Long.parseLong(getMetadataValue(dataTableMetadata, DataTable.NUM_DOCS_SCANNED_METADATA_KEY));
+      numDocsScannedInFilter = Long.parseLong(getMetadataValue(dataTableMetadata, DataTable.NUM_ENTRIES_SCANNED_IN_FILTER_METADATA_KEY));
+      numDocsScannedPostFilter = Long.parseLong(getMetadataValue(dataTableMetadata, DataTable.NUM_ENTRIES_SCANNED_POST_FILTER_METADATA_KEY));
+    } catch (NumberFormatException ignored) {
+      // best effort conversion only
```

Comment:
This should not happen. Let's at least add a warning log to catch this.

Reply:
Done.